website/docs/language/tests/index.mdx

```diff
@@ -0,0 +1,685 @@
+
+If you only use a single setup module as an alternate module, and it executes first, or you use no alternate modules, then the order of destruction will not affect you. Anything more complex may require careful consideration to make sure the destruction of created resources can complete automatically.
+
+## Expecting Failures
+
+By default, if any [Custom Conditions](/terraform/language/expressions/custom-conditions), including `check` block assertions, fail during the execution of a Terraform test file then the overall command will report the test as a failure. It is a common testing paradigm, however, to want to test failure cases. Terraform supports the `expect_failures` attribute for this use case.
+
+In each `run` block the `expect_failures` attribute can provide a list of checkable objects (resources, data sources, check blocks, input variables, and outputs) that should fail their custom conditions. The test will then pass overall if these checkable objects report an issue, while the test will fail overall if they do not report an issue.
+
+You can still write assertions alongside an `expect_failures` block, but you should be mindful that all custom conditions, except check block assertions, halt the execution of Terraform. This still applies during test execution, so your assertions should only consider values that you are sure will be computed before the checkable object is due to fail. This can be managed via references or the `depends_on` meta-argument within your main configuration.
+
+This also means that, with the exception of `check` blocks, only a single checkable object can be reliably included. We support a list of checkable objects within the `expect_failures` attribute purely for `check` blocks.
+
+A quick example here demonstrates testing the `validation` block on an input variable.
+
+```hcl
+# main.tf
+
+variable "input" {
+  type = number
+
+  validation {
+    condition = var.input % 2 == 0
+    error_message = "must by even number"
+  }
+}
+```
+
+```hcl
+# input_validation.tftest.hcl
+
+variables {
+  input = 0
+}
+
+run "zero" {
+  # The variable defined above is even, so we expect the validation to pass.
+
+  command = plan
+}
+
+run "one" {
+  # This time we set the variable is odd, so we expect the validation to fail.
+
+  command = plan
+
+  variables {
+    input = 1
+  }
+
+  expect_failures = [
+    var.input,
+  ]
+}
+```
+
+>> **Note**: Failures are only expected in the operation specified by the `command` attribute of the `run` block.
+
+This means you should be careful when using `expect_failures` in `run` blocks with `command = apply`. A `run` block with `command = apply` that expects a custom condition failure will fail overall if that custom condition fails during the plan.
```

Comment:
Starting the sentence with "this means" feels weird here, are you building off the note above? Could we do this instead?

```suggestion
Be careful when using `expect_failures` in `run` blocks with `command = apply`. A `run` block with `command = apply` that expects a custom condition failure will fail overall if that custom condition fails during the plan.
```